### PR TITLE
Merge/migrate tools/fix* programs

### PR DESCRIPTION
**Pull request by petrgazarov**
Addresses https://github.com/cdnjs/cdnjs/issues/9333.
- Consolidates `tools/fixFilenameField.js`, `tools/fixLicense.js` and `tools/fixFormat.js` into `tools/fixFormat.js`.
- removes `fixRepo.js` as it is a duplicate of `fixLicense.js`.

New to the repo, so eager to work with any feedback. Thanks!!

### CODE REVIEW

**Comment by petrgazarov** [comment]
@PeterDaveHello Ready for a review

**Comment by PeterDaveHello** [comment]
@petrgazarov Awesome! Thanks!

**Comment by PeterDaveHello** [comment]
@cdnjs/intern2 please take the first review of this change.

**Comment by pvnr0082t** [comment]
LGTM

**Comment by kennynaoh** [comment]
@petrgazarov 
The PR https://github.com/cdnjs/cdnjs/pull/9461 just merge, please fix the conflict.
Thanks!

**Comment by petrgazarov** [comment]
Rebased 🚀

**Comment by PeterDaveHello** [comment]
@cdnjs/intern2 let's review it again!

**Comment by pvnr0082t** [comment]
LGTM

**Comment by PeterDaveHello** [comment]
hmmmm ... actually, I mean to do them all in the single async.each call, which would be more efficient, @petrgazarov would you be interedted in doing that? If not, we can still merge it and do it later, but I'll prefer to do it together, thanks.

**Comment by petrgazarov** [comment]
@Yup, @PeterDaveHello. I just wasn't sure if functions were ever to be used apart from each other. Will make the change.

**Comment by PeterDaveHello** [comment]
@petrgazarov Thank you very much!

**Comment by petrgazarov** [comment]
@PeterDaveHello Done. Please check. I checked that no errors are raised but not sure about the best way to test all uses/edge cases. What do you think about adding a unit test here?

**Comment by PeterDaveHello** [comment]
@petrgazarov It would be very awesome to add aome tests in, we are really lack of testing! I could provide some test cases if you will be interested in! Thanks a lot.

**Comment by petrgazarov** [comment]
Sure, @PeterDaveHello, thanks. I'll try to get to it in the next couple days. I see you have `vows` library already in the code, is that your preference for unit testing?

**Comment by PeterDaveHello** [comment]
@petrgazarov I'm fine if you prefer to use some other test framework which you are familiar with :)

**Comment by petrgazarov** [comment]
Added tests 🚀

**Comment by PeterDaveHello** [comment]
@petrgazarov Awesome! I'll take a look at it, thanks a lot.

**Comment by PeterDaveHello** [comment]
@pvnr0082t let's verify the behavior of this change!

File: package.json [hunk 1]
```diff
@@ -9,7 +9,7 @@
     "url": "git://github.com/cdnjs/cdnjs.git"
   },
   "scripts": {
-    "test": "vows"
+    "test": "vows && jasmine"
```
Comment (PeterDaveHello) [inline package.json]: I wonder if we could separate this part of test to something like `devTest` as we use `npm test` for libraries' package.json validation for a long time, what do you think?

**Comment by petrgazarov** [comment]
Updated, @PeterDaveHello.

File: tools/fixFormat.js [hunk 2]
```diff
@@ -1,104 +1,238 @@
...
+    if (orig[orig.length - 2] !== 'min') {
+      var temp = orig,
+        ext = temp.pop();
+      temp.push('min');
+      temp.push(ext);
+      min = temp.join('.');
+    }
+    if (min !== '' && isThere('./ajax/libs/' + pkg.name + '/' + pkg.version + '/' + min)) {
+      pkg.filename = min;
+      fs.writeFileSync(item, JSON.stringify(pkg, null, 2) + '\n', 'utf8');
```
Comment (PeterDaveHello) [inline tools/fixFormat.js]: Looks like we'll write the file at line 36, so we still need to write file here? Thanks.

Comment (petrgazarov) [inline tools/fixFormat.js]: Yeah. There are a couple places where this happens. It should probably be done only once, in the end of the function.  I'll check on it.

**Comment by petrgazarov** [comment]
In the recent PR update I optimized the `writeFile`, added 3 more test examples and a hack for the fixFormat function to not be executed for real when running jasmine tests. 🎸  Let me know if this is good.

**Comment by PeterDaveHello** [comment]
@pvnr0082t please also help test this.

**Comment by pvnr0082t** [comment]
@PeterDaveHello I have tested and it can work well. :+1:

**Comment by PeterDaveHello** [comment]
@pvnr0082t Great!

**Comment by PeterDaveHello** [comment]
@petrgazarov can we remove "updated" message - `Library - react-virtualized was updated!` if its `package.json` is not changed? I thought it's the legacy part from `fixLicense.js` ... sorry about that.

**Comment by petrgazarov** [comment]
@PeterDaveHello Good call. I'm checking now with lodash `isEqual` which performs a deepdiff.

**Comment by PeterDaveHello** [comment]
@petrgazarov just noticed that the indent will not be fixed if the json content is not updated, I think we also need to stringify json to compare the indent (or you have some better ideas).

**Comment by petrgazarov** [comment]
@PeterDaveHello Yeah, great point. I would say, if logging which package json files were updated is important, this solution is probably the best bet. Normally if that's not important, I would just write over it regardless, since there isn't much of a performance win imo (considering file is read and diffed in the if condition).

**Comment by petrgazarov** [comment]
Thanks for the detailed code review btw, and if any other such tasks come your way, feel free to tag me!

**Comment by PeterDaveHello** [comment]
@petrgazarov no worry, I think it's time to merge this, we'll see :)

**Comment by PeterDaveHello** [comment]
@petrgazarov would you be interested in helping add the feature to remove the keys with leading underscore? Which are added by the new versions of npm, thanks a lot!

**Comment by PeterDaveHello** [comment]
For example: https://github.com/clairetsai818/cdnjs/blob/64eb46ae547357fcd918d01be5ebaa69b4330d9e/ajax/libs/MinPubSub/package.json#L31-L40

**Comment by petrgazarov** [comment]
@PeterDaveHello Totally!

**Comment by PeterDaveHello** [comment]
@petrgazarov Thanks a lot! (Maybe can even add this in the test!)
